packages/promise-cache/src/promiseCache.ts

```diff
@@ -7,11 +7,19 @@ export class PromiseCache<T, U> {
     this.ttl = ttlInSeconds * 1000 // Convert seconds to milliseconds
   }
 
-  async wrap(key: T, delegate: () => Promise<U>): Promise<U> {
+  async wrap(
+    key: T,
+    delegate: () => Promise<U>,
+    ttlInSeconds?: number
+  ): Promise<U> {
     const now = Date.now()
     const cached = this.cache.get(key)
```

Comment:
BTW: keys should either be case insensitive (.toLowerCase()) or we need to use url .toLowerCase() in the implementation

Reply:
I have added an optional flag to the constructor, allowing consumers to enable case sensitive keys.